atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1569,12 +1581,14 @@ protected boolean shouldDeleteAndRollback() {
     @Override
     public final void delete(TableReference tableRef, Set<Cell> cells) {
         putInternal(tableRef, Cells.constantValueMap(cells, PtBytes.EMPTY_BYTE_ARRAY));
+        cells.forEach(cell -> cache.write(tableRef, cell, CacheValue.empty()));
     }
 
     @Override
     public void put(TableReference tableRef, Map<Cell, byte[]> values) {
         ensureNoEmptyValues(values);
         putInternal(tableRef, values);
+        values.forEach((cell, value) -> cache.write(tableRef, cell, CacheValue.of(value)));
```

Comment:
didn't realise this was done in a map, I can make this just accept a map instead of individual values.

Reply:
👍